src/main/java/com/rocketden/main/controller/v1/FrontendRedirectController.java

```diff
@@ -0,0 +1,22 @@
+package com.rocketden.main.controller.v1;
+
+import org.springframework.boot.web.servlet.error.ErrorController;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.RequestMapping;
+
+// Redirects non-API routes to the frontend to avoid Spring 404 errors
+@Controller
+public class FrontendRedirectController implements ErrorController {
+
+    private static final String PATH = "/error";
+
+    @RequestMapping(value = PATH)
+    public String error() {
+        return "forward:/index.html";
+    }
+
+    @Override
+    public String getErrorPath() {
+        return PATH;
+    }
```

Comment:
I don't understand this code but it works great with the `./mvnw spring-boot:run` command, pulling up the correct API return! (It doesn't work with `npm start` in the `frontend` folder, but that's as expected I believe.) No explanation necessary, I remember you linked earlier, just wanted to say gj 👍